packages/react-native-platform-override/src/Serialized.ts

```diff
@@ -60,39 +59,43 @@ const CopyEntryType = t.type({
   issue: t.number,
 });
 
-const NonPlatformEntryType = t.union([
-  PatchEntryType,
-  DerivedEntryType,
-  CopyEntryType,
+/**
+ * Union of all serialized override types
+ */
+const OverrideType = t.union([
+  PlatformOverrideType,
+  PatchOverrideType,
+  DerivedOverrideType,
+  CopyOverrideType,
 ]);
-const EntryType = t.union([PlatformEntryType, NonPlatformEntryType]);
 
-const ManifestType = t.type({overrides: t.array(EntryType)});
+const ManifestType = t.type({overrides: t.array(OverrideType)});
 
-export type PlatformEntry = t.TypeOf<typeof PlatformEntryType>;
-export type PatchEntry = t.TypeOf<typeof PatchEntryType>;
-export type DerivedEntry = t.TypeOf<typeof DerivedEntryType>;
-export type CopyEntry = t.TypeOf<typeof CopyEntryType>;
-export type NonPlatformEntry = t.TypeOf<typeof NonPlatformEntryType>;
-export type Entry = t.TypeOf<typeof EntryType>;
+export type PlatformOverride = t.TypeOf<typeof PlatformOverrideType>;
+export type PatchOverride = t.TypeOf<typeof PatchOverrideType>;
+export type DerivedOverride = t.TypeOf<typeof DerivedOverrideType>;
+export type CopyOverride = t.TypeOf<typeof CopyOverrideType>;
+export type Override = t.TypeOf<typeof OverrideType>;
 export type Manifest = t.TypeOf<typeof ManifestType>;
 
 /**
  * Read an override manifest from a file.
  *
  * @throws if the file is invalid or cannot be found
  */
-export async function readFromFile(filePath: string): Promise<Manifest> {
+export async function readManifestFromFile(
+  filePath: string,
+): Promise<Manifest> {
   const json = (await fs.promises.readFile(filePath)).toString();
-  return this.parse(json);
+  return parseManifest(json);
 }
 
 /**
  * Parse a string with JSON for the override manifest into one.
  *
  * @throws if the JSON doesn't describe a valid manifest
  */
-export function parse(json: string): Manifest {
+export function parseManifest(json: string): Manifest {
   const parsed = JSON.parse(json);
```

Comment:
> = [](start = 14, length = 2)

You should be able to assign the type so you don't need the `as Manfiest` later

```suggestion
 : Manifest =
```

Reply:
We don't know yet whether the JSON on disk conforms to the schema. The `ManifestType.decode(parsed)` call validates against the type, which is why we delay associating the type until we know its correct.